pkg/reconciler/delivery/policy_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package delivery
+
+import (
+	"fmt"
+	"testing"
+)
+
+var (
+	pa = Policy{"time", []int{0,1,2,3,4,5,6,7,8,99}, 5}
+	pb = Policy{"request", []int{0,90,91,92,93,94,95,96,97,98,99}, 500}
+	pc = Policy{"error", []int{0, 5, 20, 50, 80, 95}, 3}
+	p0 = Policy{"time", []int{}, 10}
+	pX = Policy{"request", []int{90, 80, 70}, 5}
+	errNonNil = fmt.Errorf("It doesn't matter what this error string is")
+)
+
+func fakeCompareErrors(e1 error, e2 error) bool {
+	return (e1 == nil && e2 == nil) || (e1 != nil && e2 != nil)
+}
+
+func TestComputeNewPercent(t *testing.T) {
+	var tests = []struct {
+		policy *Policy
+		cp      int
+		want    int
+		err     error
+	}{
+		{&pa, 3, 4, nil},
+		{&pa, 50, 0, errNonNil},
+		{&pb, 99, 100, nil},
+		{&pb, 0, 90, nil},
+		{&pc, 95, 100, nil},
+		{&pc, 50, 80, nil},
+		{&pc, 100, 0, errNonNil},
+		{&p0, 0, 0, errNonNil},
+		// the last test should have undefined behavior because policy Percents field must be sorted in increasing order
+		// in unsorted situations, the result is dependent on library implementation of sort.SearchInts
+		// this test is considered passed as long as the test driver doesn't crash
+		{&pX, 90, -1, errNonNil},
+	}
+
+	for i, tt := range tests {
+		testname := fmt.Sprintf("test_%d", i)
```

Comment:
In general we try to give meaningful name to the test, rather than just index.
Then you can use the `tc.name` in the `t.Run` call.

Reply:
done